metrics/ci_integrations/test/destination/firestore/adapter/firestore_destination_client_adapter_test.dart

```diff
@@ -79,66 +90,82 @@ void main() {
+      ".addBuilds() throws an ArgumentError if a project with the given id does not exist",
       () {
-        whenFetchProject().thenThrow(GrpcError.notFound());
+        whenFetchProject().thenAnswer((_) => Future.value(_documentMock));
+        when(_documentMock.exists).thenReturn(false);
 
         final result = adapter.addBuilds(testProjectId, []);
 
-        expect(result, completes);
+        expect(result, throwsArgumentError);
       },
     );
 
     test(
-      ".addBuilds() throws an exception if the firestore throws exception different from GrpcError.notFound",
-      () {
-        whenFetchProject().thenThrow(Exception());
+      ".addBuilds() throws a DestinationError if fetching a project throws",
+      () async {
+        whenFetchProject().thenThrow(grpcError);
 
         final result = adapter.addBuilds(testProjectId, []);
 
-        expect(result, throwsException);
+        expect(result, throwsDestinationError);
       },
     );
 
     test(
-      ".addBuilds() rethrows GrpcError with status different from 'notFound'",
+      ".addBuilds() does not add builds if fetching the project throws",
       () {
-        final error = GrpcError.cancelled();
-        whenFetchProject().thenThrow(error);
+        whenFetchProject().thenThrow(grpcError);
 
         final result = adapter.addBuilds(testProjectId, []);
 
-        expect(result, throwsA(equals(error)));
+        expect(result, throwsDestinationError);
+        verifyNever(_firestoreMock.collection('build'));
       },
     );
 
     test(
-      ".addBuilds() does not add builds if fetching the project throws",
-      () async {
-        whenFetchProject().thenThrow(GrpcError.notFound());
+      ".addBuilds() does not add builds if a project with the given id does not exist",
+      () {
+        whenFetchProject().thenAnswer((_) => Future.value(_documentMock));
+        when(_documentMock.exists).thenReturn(false);
 
-        await adapter.addBuilds(testProjectId, []);
+        final result = adapter.addBuilds(testProjectId, []);
 
+        expect(result, throwsArgumentError);
         verifyNever(_firestoreMock.collection('build'));
       },
     );
 
+    test(
+      ".addBuilds() throws a DestinationError if creating a build throws",
+      () async {
+        whenFetchProject().thenAnswer((_) => Future.value(_documentMock));
+        when(_documentMock.exists).thenReturn(true);
+        when(_documentMock.id).thenReturn(testProjectId);
+        when(_firestoreMock.collection('build'))
+            .thenReturn(_collectionReferenceMock);
+        when(_collectionReferenceMock.document(
+          argThat(anyOf(testProjectIds)),
+        )).thenReturn(_documentReferenceMock);
+        when(_documentReferenceMock.create(argThat(anything)))
+            .thenAnswer((_) => Future.error(grpcError));
+
+        final result = adapter.addBuilds(testProjectId, builds);
+        await expectLater(result, throwsDestinationError);
```

Comment:
I suppose the usual `expect` would be enough here - could you please recheck? Just remove the `async` modifier and: 
```suggestion
        expect(result, throwsDestinationError);
```